### ISSUE TITLE
tab is not working in the form when sweet alert is pop up

### ISSUE THREAD

**Issue by atulintel** [t4t5/sweetalert#391]
I am using sweet alert in the html form which is popup on button click but the when i click on ok button of popup, the tab index does not work after that and i have to click each and every field to fill it.

**Comment by mfontolan** [comment]
The same think happens to me.

**Comment by krishnr** [comment]
Yup, happens to me as well. Also, hitting enter to dismiss the pop up removes the focus from the page and you cannot focus on the page again without clicking.

**Comment by fxghqc** [comment]
remove window events in doneFunction resolve this.
```javascript
swal({
  title: "Are you sure?",
  text: "You will not be able to recover this imaginary file!",
  type: "warning",
  showCancelButton: true,
  confirmButtonColor: "#DD6B55",
  confirmButtonText: "Yes, delete it!",
  closeOnConfirm: false
}, function(){

  // remove these events;
  window.onkeydown = null;
  window.onfocus = null;

  swal("Deleted!", "Your imaginary file has been deleted.", "success");
});
```

**Comment by ecwyne** [comment]
@fxghqc removing event handlers in the callback works for me. Can this be included in the main code to prevent this bug?

**Comment by henry74** [comment]
+1

**Comment by shlomogoldstein** [comment]
When will this be fixed?

**Comment by knownAsKyle** [comment]
+1

**Comment by EduardJS** [comment]
Is there any fork of this bug I can work with ? I am not sure how to use @xiaouze88's commit.

**Comment by jeffbromberger** [comment]
This bug still exists.  You can even see it on the main sweetalert demo page... http://t4t5.github.io/sweetalert/

Simply click the Try Me button for the AJAX example and then notice that tab no longer works.  I can see in the debugger that window.onfocus is not being restored to null afterwords.  If I manually set it then I can use tab again.  But setting it to null in the callback function passed to swal() does NOT work for me.  I set it to null there, but something else internal to Sweetalert is setting it back before the whole process ends.  Breaking into the debugger several seconds later confirms this.

Please fix!

**Comment by mhoffmeyerDC** [comment]
I can confirm this behavior as well. It's intermittent, the best type of bug...
I am seeing the `window.onkeydown` event being bound to the function even after dialogs have closed... and possibly even before a dialog has been triggered.

**Comment by Mr-Anonymous** [comment]
I am experiencing the exact same issue as well. When the SweetAlert pop-up is closed, tabs in the page's form stops working even with `tabindex` set.

**Comment by EduardJS** [comment]
I will make a PR tomorrow with a fix for this.

**Comment by Mr-Anonymous** [comment]
That will be really great @EduardJS Thank you for that.

For now I am using the solution from @JustinWinthers in another thread:

```
(function (){

    var _swal = window.swal;

    window.swal = function(){

        var previousWindowKeyDown = window.onkeydown;

        _swal.apply(this, Array.prototype.slice.call(arguments, 0));

        window.onkeydown = previousWindowKeyDown;

    };

})();
```
REF: https://github.com/t4t5/sweetalert/issues/127

**Comment by edwin1217** [comment]
@Mr-Anonymous 
update

> var _swalclose=window.swal.close;
	var _swal = window.swal;
	window.swal = function(){
		var previousWindowKeyDown = window.onkeydown;
		_swal.apply(this, Array.prototype.slice.call(arguments, 0));
		window.onkeydown = previousWindowKeyDown;
	};
	window.swal.close=function(){
		_swalclose.apply(this);
	};

**Comment by elasticsteve** [comment]
Wow! I just found this thread. I spent days on wondering why my tabs work and then they don't. Now I realize that sometimes I use a sweetalert and that's the reason it stopped!

Is anyone still working on a fix for this and what's the best solution to deal with it for now?

**Comment by EduardJS** [comment]
@elasticsteve If you pay attention to the other threads on this repo you will notice this one is not maintained anymore.

There is another repo called [SweetAlert2](https://limonte.github.io/sweetalert2/), which you should really port your apps towards.

If you still want to use this version though, the only solution is this one. I forgot to make a PR.. didn't have enough time. @edwin1217's solution is the one you should go with.

**Comment by kosst** [comment]
@edwin1217's solution is nearly the same i have posted in thread #127 a while ago.

**Comment by elasticsteve** [comment]
Thanks everyone. I tried to port over to SweetAlert2, but had to give up. I couldn't make a simple text input with `closeOnConfirm: false` work. After wasting 3 hours I reverted back and will try the suggestions above.

**Comment by elasticsteve** [comment]
@kosst Yes, your solution is much better, because with the one from @edwin1217 the default Enter key action for OK and error messages (input text modals) wouldn't work.

Here again what's working for me best:
```
(function (){
    var close = window.swal.close;
    window.swal.close = function() {
        close();
        window.onkeydown = null;
    };
})();
```

**Comment by Omniscience619** [comment]
> }

I know it's been a really long time, but would you be kind enough to explain as to how I can call this function as a whole in another function?  Where should I place this snippet in controller?

function(isConfirm){  //Function that triggers on user action.
        if(isConfirm){
          $scope.myData.splice(index,1);
          SweetAlert.swal("Deleted!");
        } else {
            SweetAlert.swal("Your file is safe!");
        }

**Comment by khoiwalvikram2** [comment]
change - 
closeOnConfirm: false
TO - 
closeOnConfirm: true

that's it... it worked for me

**Comment by Trend20** [comment]
Hi everyone, I know it has been long but did any of you found a solution for this? I am still experiencing the same problem with Angular 12.
